external/implementations/samlconf-openam-impl/README.md

```diff
@@ -0,0 +1,58 @@
+<!--
+Copyright (c) 2018 Codice Foundation
+
+Released under the GNU Lesser General Public License; see
+http://www.gnu.org/licenses/lgpl.html
+-->
+
+# The [OpenAM](https://github.com/OpenIdentityPlatform/OpenAM) SAML Implementation
+## Known Compliance Issues
+| Issue| Section | Specification Snippet
+| -----| ------- | ---------------------
+| Does not use POST messages as the action attribute | Bindings 3.5.4 | The action attribute of the form MUST be the recipient's HTTP endpoint for the protocol or profile using this binding to which the SAML message is to be delivered. The method attribute MUST be "POST". 
+| Does not validate that Relay States are less than or equal to 80 bytes.| Bindings 3.4.3 | RelayState data MAY be included with a SAML protocol message transmitted with this binding. The value MUST NOT exceed 80 bytes.
+| Does not validate the Name ID in an AuthnRequest | Profiles 4.1.4.1 | SAMLProfiles_4_1_4_1: Note that the service provider MAY include a <Subject> element in the request that names the actual identity about which it wishes to receive an assertion. This element MUST NOT contain any <SubjectConfirmation> elements. If the identity provider does not recognize the principal as that identity, then it MUST respond with a <Response> message containing an error status and no assertions.
+| Does not sign Logout Responses | Profiles 4.4.4.2 | The responder MUST authenticate itself to the requester and ensure message integrity, either by signing the message or using a binding-specific mechanism.
+| Does not use the same given NameID for Logout Request | Profiles 4.4.4.1 | The principal MUST be identified in the request using an identifier that strongly matches the identifier in the authentication assertion the requester issued or received regarding the session being terminated, per the matching rules defined in Section 3.3.4 of [SAMLCore].
+| (Does not properly use RelayStates) | Bindings 3.5.4 | If a "RelayState" value is to accompany the SAML protocol message, it MUST be placed in an additional hidden form control named RelayState within the same form with the SAML message.
+| (Does not properly use Relay States) | Bindings 3.4.3 | If a SAML request message is accompanied by RelayState data, then the SAML responder MUST return its SAML protocol response using a binding that also supports a RelayState mechanism, and it MUST place the exact data it received with the request into the corresponding RelayState parameter in the response.
```

Comment:
📘 I wasn't too sure on what to put under the Issue section for these. Any ideas?

Reply:
They document relay states here: https://backstage.forgerock.com/docs/am/5/saml2-guide/#using-saml2-sso-slo in the forgerock docs (which are leaps and bounds above the mirror'd docs from OpenIdentityProvider/OpenAm)

Reply:
I ended up removing the second RelayState error since that only appeared after I commented out the original check that the relay state was not sent